openshift-with-appstudio-test/e2e/jvm_build_service_test.go

```diff
@@ -6,331 +6,129 @@ package e2e
-	}
-	defer resp.Body.Close()
-	bytes, err := io.ReadAll(resp.Body)
-	if err != nil {
-		debugAndFailTest(ta, err.Error())
-	}
-	return decodeBytesToTektonObjbytes(bytes, obj, ta)
-}
 
-func streamFileYamlToTektonObj(path string, obj runtime.Object, ta *testArgs) runtime.Object {
-	bytes, err := ioutil.ReadFile(path)
-	if err != nil {
-		debugAndFailTest(ta, err.Error())
-	}
-	return decodeBytesToTektonObjbytes(bytes, obj, ta)
-}
+	ta.t.Run("build-container task from component pipelinerun references a correct analyzer image", func(t *testing.T) {
+		ciAnalyzerImage := os.Getenv("JVM_BUILD_SERVICE_ANALYZER_IMAGE")
 
-func TestExampleRun(t *testing.T) {
-	ta := setup(t, nil)
-	//TODO, for now at least, keeping our test project to allow for analyzing the various CRD instances both for failure
-	// and successful runs (in case a run succeeds, but we find something amiss if we look at passing runs; our in repo
-	// tests do now run in conjunction with say the full suite of e2e's in the e2e-tests runs, so no contention there.
-	//defer projectCleanup(ta)
-
-	path, err := os.Getwd()
-	if err != nil {
-		debugAndFailTest(ta, err.Error())
-	}
-	ta.Logf(fmt.Sprintf("current working dir: %s", path))
+		if ciAnalyzerImage == "" {
+			t.Skip("JVM_BUILD_SERVICE_ANALYZER_IMAGE env var is not exported, skipping the test...")
+		}
 
-	mavenYamlPath := filepath.Join(path, "..", "..", "deploy", "base", "maven-v0.2.yaml")
-	ta.maven = &v1beta1.Task{}
-	obj := streamFileYamlToTektonObj(mavenYamlPath, ta.maven, ta)
-	var ok bool
-	ta.maven, ok = obj.(*v1beta1.Task)
-	if !ok {
-		debugAndFailTest(ta, fmt.Sprintf("file %s did not produce a task: %#v", mavenYamlPath, obj))
-	}
-	// override images if need be
-	analyserImage := os.Getenv("JVM_BUILD_SERVICE_ANALYZER_IMAGE")
-	if len(analyserImage) > 0 {
-		ta.Logf(fmt.Sprintf("PR analyzer image: %s", analyserImage))
-		for _, step := range ta.maven.Spec.Steps {
-			if step.Name != "analyse-dependencies" {
-				continue
+		err = wait.PollImmediate(ta.interval, ta.timeout, func() (done bool, err error) {
+			pr, err := tektonClient.TektonV1beta1().PipelineRuns(ta.ns).Get(context.TODO(), ta.run.Name, metav1.GetOptions{})
+			if err != nil {
+				ta.Logf(fmt.Sprintf("get pr %s produced err: %s", ta.run.Name, err.Error()))
+				return false, nil
 			}
-			ta.Logf(fmt.Sprintf("Updating analyse-dependencies step with image %s", analyserImage))
-			step.Image = analyserImage
-		}
-	}
-	sidecarImage := os.Getenv("JVM_BUILD_SERVICE_SIDECAR_IMAGE")
-	if len(sidecarImage) > 0 {
-		ta.Logf(fmt.Sprintf("PR sidecar image: %s", sidecarImage))
-		for _, sidecar := range ta.maven.Spec.Sidecars {
-			if sidecar.Name != "proxy" {
-				continue
+
+			for _, tr := range pr.Status.TaskRuns {
+				if tr.PipelineTaskName == "build-container" && tr.Status != nil && tr.Status.TaskSpec != nil && tr.Status.TaskSpec.Steps != nil {
+					for _, step := range tr.Status.TaskSpec.Steps {
+						if step.Name == "analyse-dependencies-java-sbom" {
```

Comment:
fyi @brunoapimentel here is the other side the equation, the check we talked about, where we verify the bundle override you built is getting used :-)